web/src/ApiButton.tsx

```diff
@@ -368,6 +377,37 @@ export function ApiButton(props: React.PropsWithChildren<ApiButtonProps>) {
 
   const disabled = loading || uiButton.spec?.disabled
 
+  if (confirming) {
+    return (
+      <ApiButtonRoot
```

Comment:
the disadvantage of returning a different dom node for the confirmation state is that in privileges mouse users. if i'm interacting with the disable button using a keyboard, then that button disappears and i'm no longer focused on it. to confirm or dismiss, i have to navigate back in the tab order. if i'm using an assistive device, there's no immediate (non-visual) indication that i need to confirm the action i just tried to perform.

ideally, (i think) the enable/disable button is always rendered inside a button group, and just the button text changes when you click `disable`. the `cancel` button is added to the button group during the confirmation state. the `i think` part is mostly because i can't quickly/cleanly map this "button with a confirmation and cancel state" to any of the aria docs examples. there is an [`alertdialog` role](https://www.w3.org/TR/wai-aria-1.1/#alertdialog), which indicates to the user that there's an alert that requires their attention/interaction, so that could be useful on the button group level (but then that overrides the mui component's `group` role). 🤔 

a workaround for the code as-is might be to manually manage focus in the click handler, so entering the confirmation state focuses on the confirmation button or button group. but that may end up being hacky bc that element doesn't exist on the dom just yet.

...ok, end of mini-essay! with all that being said, i don't think this is a blocker for this pr, but it's something i think we should look into (perhaps as a side ticket in bulk actions?). do you have any thoughts? you may have tried some of this stuff already and gone with this approach because of other things i'm not considering!

Reply:
This all sounds right and is not a thing I was considering. I agree this isn't a blocker for this pr and we can tackle it later.